lib/src/layer/polyline_layer.dart

```diff
@@ -9,7 +12,16 @@ import 'package:flutter_map/src/misc/offsets.dart';
 import 'package:flutter_map/src/misc/point_extensions.dart';
 import 'package:latlong2/latlong.dart';
 
-class Polyline {
+class _LastHit<TapKeyType extends Object> {
+  ({List<Polyline<TapKeyType>> tapped, LatLng point})? hit;
+}
+
+typedef PolylineOnTap = void Function(LatLng point);
+typedef PolylineLayerOnTap<TapKeyType extends Object> = void Function(
+    List<TapKeyType> tappedKeys, LatLng point)?;
```

Comment:
Could we use List<Polygon> here? This would give users more flexibility.

Reply:
If we just pass around the `Polyline` instead of the key, then we don't need the `TapKeyType` at all - users can just use an `Object`.
I was concerned about the performance of passing the entire object along everywhere, but I guess Dart is smart enough to pass by reference. I'll replace it.

Reply:
Resolved.

Reply:
Both is just a pointer that gets passed on so it shouldn't make a difference in performance.
(As I learned from you) `TapKeyType` is still required that the Polyline gets recognized as interactive by the layer. 
But in case other properties of the polygon need to be accessed I think it still makes sense to have List<Polyline> returned. 🤔

Reply:
Yep, that's how I've implemented it.